internal/backend/azure/azure.go

```diff
@@ -245,12 +276,15 @@ func (be *Backend) saveSmall(ctx context.Context, objName string, rd backend.Rew
 		return errors.Wrap(err, "StageBlock")
 	}
 
+	commitBlockListOptions := &blockblob.CommitBlockListOptions{}
+	commitBlockListOptions.Tier = &accessTier
+
 	blocks := []string{id}
-	_, err = blockBlobClient.CommitBlockList(ctx, blocks, &blockblob.CommitBlockListOptions{})
+	_, err = blockBlobClient.CommitBlockList(ctx, blocks, commitBlockListOptions)
```

Comment:
I can't make this a GH suggestion because it spans a non-altered line, but this can be shortened to

```go
	blocks := []string{id}
	_, err = blockBlobClient.CommitBlockList(ctx, blocks, &blockblob.CommitBlockListOptions{
		Tier: &accessTier,
	})
```